crates/bevy_render/src/texture/image_texture_conversion.rs

```diff
@@ -1,194 +1,233 @@
 use crate::texture::{Image, TextureFormatPixelInfo};
+use anyhow::anyhow;
 use image::{DynamicImage, ImageBuffer};
 use wgpu::{Extent3d, TextureDimension, TextureFormat};
 
-// TODO: fix name?
-/// Converts a [`DynamicImage`] to an [`Image`].
-pub(crate) fn image_to_texture(dyn_img: DynamicImage, is_srgb: bool) -> Image {
-    use bevy_core::cast_slice;
-    let width;
-    let height;
-
-    let data: Vec<u8>;
-    let format: TextureFormat;
-
-    match dyn_img {
-        DynamicImage::ImageLuma8(i) => {
-            let i = DynamicImage::ImageLuma8(i).into_rgba8();
-            width = i.width();
-            height = i.height();
-            format = if is_srgb {
-                TextureFormat::Rgba8UnormSrgb
-            } else {
-                TextureFormat::Rgba8Unorm
-            };
-
-            data = i.into_raw();
-        }
-        DynamicImage::ImageLumaA8(i) => {
-            let i = DynamicImage::ImageLumaA8(i).into_rgba8();
-            width = i.width();
-            height = i.height();
-            format = if is_srgb {
-                TextureFormat::Rgba8UnormSrgb
-            } else {
-                TextureFormat::Rgba8Unorm
-            };
-
-            data = i.into_raw();
-        }
-        DynamicImage::ImageRgb8(i) => {
-            let i = DynamicImage::ImageRgb8(i).into_rgba8();
-            width = i.width();
-            height = i.height();
-            format = if is_srgb {
-                TextureFormat::Rgba8UnormSrgb
-            } else {
-                TextureFormat::Rgba8Unorm
-            };
-
-            data = i.into_raw();
-        }
-        DynamicImage::ImageRgba8(i) => {
-            width = i.width();
-            height = i.height();
-            format = if is_srgb {
-                TextureFormat::Rgba8UnormSrgb
-            } else {
-                TextureFormat::Rgba8Unorm
-            };
-
-            data = i.into_raw();
-        }
-        DynamicImage::ImageLuma16(i) => {
-            width = i.width();
-            height = i.height();
-            format = TextureFormat::R16Uint;
+impl Image {
```

Comment:
I don't understand why this was turned into an impl block. It makes the diff much harder to read and I'm not sure it really adds anything.

Reply:
I completely agree that adding the impl makes the code less pleasant. The reason I ended up including it in the first place was to list the functions on `Image`'s doc page to make them easier to find because that's where I went looking to find something to help me edit image data.

As for why I didn't just impl `From`/`TryInto` on `Image` (which would have been much cleaner and also provided `Into<Image>` for `DynamicImage`) and use the functions internally:
```
impl From<DynamicImage> for Image {
	fn from(img: DynamicImage) -> Self {
		image_to_texture(image, x) // x is probably true right?
	}
}
```
By default the user would be forced to use srgb (which might be a good thing I don't know).

Which of the following would you like me to do:
1. Move the methods into the impl in `image.rs` and delete `image_texture_conversion.rs`
2. Set the functions as private, use std::convert traits, and set some hardcoded value for `is_srgb`.
3. Keep things as they are.
4. Something else...

By the way, I am in no way an expert on anything so there might be something I'm missing here. Constructive criticizm is welcome.

Reply:
@IceSentry you can toggle "show whitespace" in github at the top of the file diff, it's one of the buttons under the "Files Changed" tabs.

I honestly think it makes more sense as a `impl Image` block, since those are operation on the `Image` struct IMO, they do not make sense as free functions.

Reply:
@nicopap I thought I already had it turned on, apparently github decided to turn it off 😢. The diff is way cleaner now. It originally looked like everything was moved around and it seemed a bit too much. Sorry about that.

The docs discoverability is a good point I hadn't considered, I shouldn't have reviewed late at night.

@micron-mushroom  For this PR I think you should keep things like this. We can consider moving it or implementing more traits in a future PR